eman2/protocols/protocol_autopick_sparx.py

```diff
@@ -50,7 +50,7 @@ def __init__(self, **kwargs):
     # --------------------------- DEFINE param functions ----------------------
     def _defineParams(self, form):
         ProtParticlePickingAuto._defineParams(self, form)
-        form.addParam('boxSize', IntParam, default=100,
+        form.addParam('boxSize', IntParam, default=100, allowsPointers=True,
```

Comment:
Well, the user can still just type the number. The new is just a magnifier next to the parameter's box that can be clicked or not, as user prefer. I don't see the drawback for the user...

On the other side, in automatic workflows where protocols returning estimated boxsize can be used, it may be useful. Right, you can get the same from the API, but the simplest way to (automatic or semi-automatic) generate workflows is to use json templates. Thus, this becomes a need for linking the boxsize estimator to the Sparx protocol. Not in all facilities is able to code programs using the API, but launching workflows from templates is very simple and useful (also using the 'scipion_demo' approach where a template is followed).

Reply:
David, there is something called "status quo", so I guess the point here is not "that there is not drawback for the user" (that I agree on), the question is if there is an "advantage" for the user that justifies changes and new logic that need to be explained. In the case of json workflows, the allowsPointers is irrelavant, so this argument related to API vs json does not make much sense.

Reply:
Under my point of view, there is nothing new to be explained. The box to type the value still there with the same appearance and the vast majority will not realize that a lens is added next to it. Additionally the behavior of clicking that lens is so known in Scipion.

Regarding the json, well, if you manually write/edit the json you're right. However, I'm focus on the microscope managers that don't have much computing skills and, then, the exporting/importing projects to run/schedule workflows is very straightforward.

Reply:
I only see benefits and possibilities with this change.

Reply:
@pconesa, Sometimes "more" is not "better"...the noisy can hide the true information in some cases. ;)

Reply:
;-)

Reply:
@dmaluenda, although clicking lens is known in Scipion, it is not known the use of this feature to write a simple number. So, you are adding a new logic (and, off course, introducing more code to be maintained and source of errors) when there are other very simple ways to do it. I disagree with microscope managers' point of view. Im sure image processing is very familiar for these people and their computer skills is more than enough to manage scipion. They only need a very good documentation. 
@pconesa, you are not measuring the consequences (in terms of maintenance and errors) that this change could provoke. Also, you are breaking the homogeneity of this parameter among the different packages and this could be a source of errors.

Reply:
@josuegbl we are not discussing here the functionality of allowPointers in Scalar values. This is already approved and testes and, obviously it has the risk of adding any functionality to existing code (inherent to any development activity)

What is discussed here is the benefit of "allowing" to use a pointer in the boxSize (which in anycase it will be optional). 

I'm reducing the noise now.

Reply:
@pconesa I understood what is discussed here. Both points  are closely related cause boxSize is a Scalar... oh, coincidence! In any case, I suggest that the maintainer of this plugging take a desicion to allow this or not. Is his fully competence.

Reply:
No, you don't fully understand it, it is a scalar, and it we still be an Scalar, there is no need to change any code and should work

Reply:
I agree with you, that the maintainers have the last word.

Reply:
All the code for this feature was in Scipion before this PR and it will be there because it was consensued and approved. And it was approved because we find a way to keep the compatibility and the homogeneity.

I'm not discussing the image processing skills of the microscope managers that fore sure are better than me, neither the computing skills, but it's obvious that the simplest procedure for them is to generate a project using the Scipion GUI connecting protocols, export the JSON as a templete and, then, loading that JSON to run the whole workflow (without write/edit any line of code by the microscope people). You are proposing to modify the JSON in such a way to connect the boxSize estimator with the Sparx, and this is not obvious for all people. Moreover using the Scipion API with the documentation that we have....

But if you don't want to understand it for some reason, ok, nevermind...

Reply:
Wow, quite a discussion here. I do not mind the change since you can still type a number. As long as users won't start pressing Browse button and finding no objects because they never knew about the existence of boxSizeEstimator protocol. Then they might be confused.